test/e2e/kubectl.go

```diff
@@ -201,7 +201,7 @@ var _ = framework.KubeDescribe("Kubectl alpha client", func() {
 		})
 
 		AfterEach(func() {
-			framework.RunKubectlOrDie("delete", "cronjobs", cjName, nsFlag)
+			framework.RunKubectlOrDie("delete", "cronjob.v2alpha1.batch", cjName, nsFlag)
```

Comment:
This should have worked against any version. Why was pinning needed?

Reply:
Is delete not using the priority restmapper?

Reply:
It seems 1.8 API server have multiple versions of cronjob and will return error if kubectl doesn't explicitly specify GVK of cronjob
```
{  cronjobs} matches multiple kinds [batch/v1beta1, Kind=CronJob batch/v2alpha1, Kind=CronJob
```

Reply:
I found that we already did that in master https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L237

Reply:
this is not the right fix… all kubectl resource selection should use the priority order indicated by the server and should allow the same resource in different groups and versions

Reply:
Asked about it in https://github.com/kubernetes/kubernetes/commit/6962427b35861eace9796214be85e0188e241ce6#commitcomment-24183968 as well

Reply:
That's a client side error, not server-side. We should fix the bug in delete, not mask test failures

Reply:
found the bug. the discovery rest mapper doesn't set up priority correctly when a resource exists only in non-preferred versions of a group

batch prefers v1 (which does not contain CronJobs)

Reply:
fix in https://github.com/kubernetes/kubernetes/pull/52227